### PR TITLE
Fix labels in mediation analysis

### PR DESCRIPTION
**Pull request by Kucharssim**
Fixes https://github.com/jasp-stats/jasp-issues/issues/1128

And extends unit tests

### CODE REVIEW

File: R/mediationanalysis.R [hunk 1]
```diff
@@ -340,26 +341,44 @@ MediationAnalysis <- function(jaspResults, dataset, options, ...) {
...
+  # Fill total effects
   pe_tot <- pe[pe$op == ":=" & substr(pe$lhs, 1, 3) == "tot",]
-  
-  tottab[["lhs"]]      <- rep(options$predictor, length(options$dependent))
+
+  # get predictors, outcome combinations
+  terms <- strsplit(pe_tot[["label"]], "_")
+  termsCombinations <- list(
+    x = as.integer(gsub("x", "", sapply(terms, "[[", 2))),
+    y = as.integer(gsub("y", "", sapply(terms, "[[", 3)))
```
Comment (vandenman) [inline R/mediationanalysis.R:376]: Could you use `vapply` here instead of `sapply`? Same on lines 350-352 and 422-423.

**Review by vandenman — CHANGES_REQUESTED** [review]
The issue is fixed, one minor comment about the code.

File: R/mediationanalysis.R [hunk 2]
```diff
@@ -340,26 +341,44 @@ MediationAnalysis <- function(jaspResults, dataset, options, ...) {
...
-  indtab[["x"]]        <- rep(options$predictor, each = length(options$mediators) * length(options$dependent))
+  # get predictors, mediators, outcome combinations
+  terms <- strsplit(pe_ind[["label"]], "_")
+  termsCombinations <- list(
+    x = .medGetTermsFromLavaanTable(terms, 2, "x"),
+    m = .medGetTermsFromLavaanTable(terms, 3, "m"),
+    y = .medGetTermsFromLavaanTable(terms, 4, "y")
+  )
+
+  indtab[["x"]]        <- .unv(options[["predictor"]][termsCombinations[["x"]]])
```
Comment (vandenman) [inline R/mediationanalysis.R:355]: is `.unv` necessary here? This should be done by default since automatic column encoding was enabled.

Comment (Kucharssim) [reply R/mediationanalysis.R]: Yeah you are right...

**Review by vandenman — APPROVED** [review]

**Pull request merged**